geometry-schema.json

```diff
@@ -1,91 +1,224 @@
-                }
+        "coordinates": {
+          "type": "array",
+          "items": {
+            "type": "array",
+            "minItems": 4,
+            "items": {
+              "type": "array",
+              "minItems": 2,
+              "items": {
+                "type": "number"
+              }
             }
+          }
+        },
+        "bbox": {
+          "type": "array",
+          "minItems": 4,
+          "items": {
+            "type": "number"
+          }
         }
-    ],
-    "definitions": {
-        "position": {
-            "description": "A single position",
+      }
+    },
+    {
+      "$id": "#MultiPoint",
+      "title": "GeoJSON MultiPoint",
+      "type": "object",
+      "required": [
+        "type",
+        "coordinates"
+      ],
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "MultiPoint"
+          ]
+        },
+        "coordinates": {
+          "type": "array",
+          "items": {
             "type": "array",
             "minItems": 2,
-            "items": [ { "type": "number" }, { "type": "number" } ],
-            "additionalItems": false
+            "items": {
+              "type": "number"
+            }
+          }
         },
-        "positionArray": {
-            "description": "An array of positions",
-            "type": "array",
-            "items": { "$ref": "#/definitions/position" }
+        "bbox": {
+          "type": "array",
+          "minItems": 4,
+          "items": {
+            "type": "number"
+          }
+        }
+      }
+    },
+    {
+      "$id": "#MultiLineString",
+      "title": "GeoJSON MultiLineString",
+      "type": "object",
+      "required": [
+        "type",
+        "coordinates"
+      ],
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "MultiLineString"
+          ]
         },
-        "lineString": {
-            "description": "An array of two or more positions",
-            "allOf": [
-                { "$ref": "#/definitions/positionArray" },
-                { "minItems": 2 }
-            ]
+        "coordinates": {
+          "type": "array",
+          "items": {
+            "type": "array",
+            "minItems": 2,
+            "items": {
+              "type": "array",
+              "minItems": 2,
+              "items": {
+                "type": "number"
+              }
+            }
+          }
         },
-        "linearRing": {
-            "description": "An array of four positions where the first equals the last",
-            "allOf": [
-                { "$ref": "#/definitions/positionArray" },
-                { "minItems": 4 }
-            ]
+        "bbox": {
+          "type": "array",
+          "minItems": 4,
+          "items": {
+            "type": "number"
+          }
+        }
+      }
+    },
+    {
+      "$id": "#MultiPolygon",
```

Comment:
internal $id is not used in the official schema, the rather provide a specific document per type to point directly to it.